### ISSUE TITLE
[BUG] Android 10 modules wont work untill click options optimize with dexopt

### ISSUE THREAD

**Issue by Jasi2169** [ElderDrivers/EdXposed#447]
**什么情况/What happened?**

如无法开机/模块无效等

**Xposed模块列表/Xposed Module List**

允许截图/Screenshot allowed

**Magisk模块列表/Magisk Module List**

允许截图/Screenshot allowed

**EdXposed及Riru版本/Versions of EdXposed and Riru**

EdXposed:

Riru:

**相关Logcat/Logcat**

有助于定位问题，必须使用我们的log抓取模块抓取/It can help us to locate issue, must use our logcat module

I am using build 4497,

Guys have reported me that module is not active, actually xposed doesnot hook anything even it says active after reboot, the only thing which make it work on samsung stock which i know dont know about others, you have to click option OPTIMIZE ALL APPS WITH DEXOPT in edxposed manager,after reboot now module will active and will be able to hook.

If you clear cache like in twrp and reboot they will unable to detect xposed again, so you have to use that option again

**Comment by drewcu** [comment]
What version of EdXposed Manager are you using? I had similar issues with version 4.5.4+ but no issues with 4.5.3 which I have reverted back to. In my case, over time the framework would become inactive with 4.5.4.0.1 and clearing cache would sometimes fix that but only temporarily. Modules would work in the background but if I opened SudoHide, it would complain about the framework not being active.

Had to uninstall all things EdXposed including the Manager, and install EdXposed RIRU and Sandhook 4463 from Magisk along with Manager 4.5.3.  Then I upgraded to Sandhook 4497 and cleared data from the Play Store. Now all is well.

**Comment by Jasi2169** [comment]
I am using 4.5.4 i will try to downgrade to v4.5.3 now and see

**Comment by j-remy11** [comment]
I was having this issue as well where it will says its installed but not active using the latest canary version. Downgrading to 4.5.3 seems to fixed my issue. It was working last time and then it just stopped working so I'm hoping this fixes it. Pixel 3 on Android 10

**Comment by drewcu** [comment]
4.5.3 is just a placebo I think. Over time, I find that some of my apps like Sudohide and Minminguard complain that they cannot find the framework (though they still hide / block apps) but EM 4.5.3 still says the framework is active. Clearing cache fixes this but just temporarily for me,

I have just given up and upgraded to 4.5.4 which "tells it like it is" and says the framework is inactive when it truly isn't for these apps. As long as my enabled apps still work, I guess I'm fine with this setup. Sudohide won't let me change settings without the "enabled framework" but it still hides Youtube Vanced from the Store like I need it to and I don't need to hide anything else right now. 

Now I have enabled Speed Mode and my phone is way LESS laggy than it was before and I'll take the speed tradeoff over the green Active banner in EM 4.5.4 (yellow now) and having the ability to modify Sudohide (which I don't care about).  Would be nice to have this fixed some time but I guess it's not that big a deal - more just annoying and I know the clear cache remedy should I need to modify these troublesome apps .

**Comment by JohnGLFour** [comment]
I am on Canary Magisk, on EDX v0.4.6.1 Canary (4497?), and I agree @drewcu . Everything he said was true (without Speed Mode) and it is so annoying.

I got the Module logs.
Similar to ElderDrivers/EdxposedManager#23 but mine now hooks, before nope.

**Comment by andrizmitnick** [comment]
Up

**Comment by MlgmXyysd** [comment]
At present, it is not clear whether dexopt can work normally on Android Q, please use speed (dex2oat) mode

**Comment by Jasi2169** [comment]
For me speed one didnot work only once i tried then i try dexopt all apps normal one and all.modules starts working.
